clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/evaluation/TriggersMatcher.kt

```diff
@@ -0,0 +1,404 @@
+     * within that event are met, the function returns `true`.
+     *
+     * @param whenTriggers A list of event triggers with conditions to match against the event.
+     * @param eventName The name of the event to be matched.
+     * @param eventProperties A map of event properties where keys are property names and
+     *        values are property values.
+     * @return `true` if any event matches, and all conditions
+     * within that event are met, `false` otherwise.
+     */
+    fun matchEvent(
+        whenTriggers: List<TriggerAdapter>,
+        event: EventAdapter
+    ): Boolean {
+
+        // events in array are OR-ed
+        //val event = EventAdapter(eventName, eventProperties)
+        // Check if any TriggerAdapter in the list matches the event
+        return whenTriggers.any { match(it, event) }
+    }
+
+    /**
+     * Matches a charged event against a set of trigger conditions.
+     *
+     * This function evaluates the trigger conditions for a charged event and returns `true`
+     * if all conditions within any of the charged events are met. The events in the `whenTriggers`
+     * array are checked in an OR-ed manner, meaning that if any charged event with all conditions
+     * within that event are met, the function returns `true`.
+     *
+     * @param whenTriggers A JSON array of event triggers with conditions to match against the charged event.
+     * @param eventName The name of the charged event to be matched.
+     * @param details A map of event details or properties where keys are property names and
+     *        values are property values.
+     * @param items A list of items associated with the charged event. Each item is represented as
+     *        a map of properties where keys are property names and values are property values.
+     * @return `true` if any event matches, and all conditions
+     * within that event are met, `false` otherwise.
+     */
+    fun matchChargedEvent(
+        whenTriggers: JSONArray,
+        eventName: String,
+        details: Map<String, Any>,
+        items: List<Map<String, Any>>,
+        userLocation: Location? = null
+    ): Boolean {
+
+        // events in array are OR-ed
+        val event = EventAdapter(eventName, details, items, userLocation)
+        return (0 until whenTriggers.length())
+            .map { TriggerAdapter(whenTriggers[it] as JSONObject) }
+            .any { match(it, event) }
+    }
+
+    /**
+     * Helper function to match all trigger conditions against an event.
+     *
+     * @param trigger The [TriggerAdapter] having trigger condition to be matched against the event.
+     * @param event The [EventAdapter] having event to be matched against the trigger condition.
+     * @return `true` if all conditions within the trigger condition are met, `false` otherwise.
+     */
+    @VisibleForTesting
+    internal fun match(trigger: TriggerAdapter, event: EventAdapter): Boolean {
```

Comment:
rathert than the func the entire class should be internal right?

Reply:
@piyush-kukadiya

Reply:
tech debt